lnclient/lnd/lnd.go

```diff
@@ -448,14 +415,89 @@ func NewLNDService(ctx context.Context, lndAddress, lndCertHex, lndMacaroonHex s
 		return nil, err
 	}
 
-	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey}
+	lndCtx, cancel := context.WithCancel(ctx)
+
+	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey, cancel: cancel}
+
+	// Subscribe to payments
+	go func() {
+		paymentStream, err := lndClient.SubscribePayments(lndCtx, &routerrpc.TrackPaymentsRequest{
+			NoInflightUpdates: true,
+		})
+		if err != nil {
+			logger.Logger.Errorf("Error subscribing to payments: %v", err)
+			return
+		}
+		for {
+			payment, err := paymentStream.Recv()
+			if lndCtx.Err() == context.Canceled {
```

Comment:
hmm, can you select on context.Done() instead?

Reply:
I tried adding 
```go
select {
case <-lndCtx.Done():
	return
default:
	payment, err := paymentStream.Recv()
	if err != nil {
		logger.Logger.Errorf("Failed to receive payment: %v", err)
	...
```

But it still prints the error
`{"level":"error","msg":"Failed to receive payment: rpc error: code = Canceled desc = context canceled","time":"2024-07-17T15:55:25+05:30"}`
on context cancelling

Reply:
Is it just an error log? there's no issue, right? I think we should still add this switch checking for ctx.Done.

Reply:
Yeah I'll maybe add it, select is used like you said also in lndhub.go too ([here](https://github.com/getAlby/lndhub.go/blob/main/lib/service/invoicesubscription.go#L243-L248)) 🤔 But I'm wondering because the context done should trigger the select above it instead of resulting in this, I think it's because we do this in default and by the time context is killed it goes till the execution of `paymentStream.Recv()`